subsys/net/ip/net_if.c

```diff
@@ -1733,6 +1733,7 @@ struct net_if_mcast_addr *net_if_ipv6_maddr_add(struct net_if *iface,
 		}
 
 		ipv6->mcast[i].is_used = true;
+		ipv6->mcast[i].is_joined = true;
```

Comment:
This should not be needed here as the flag is set after we have actually joined to the group in `net_ipv6_mld_join()` which adds the address to the group and calls `net_if_ipv6_maddr_join()`.

Reply:
Example samples\net\sockets\coap_server\src\coap-server.c has no call to net_ipv6_mld_join() and not working as expected after updating to v2.4.0

Reply:
Just adding multicast address to the group is not the same as joining because in that we do not send the join message in that case. The coap-server should be changed in this case to properly join the group if it needs to do that.